### PR TITLE
prov/gni: added test for 1349

### LINKED ISSUE #1352: Add test for #1349
See PR #1349

### PR DESCRIPTION
**Pull request by jswaro**
Added test for #1349 to test race conditions with authorization
keys

Signed-off-by: James Swaro <jswaro@cray.com>

closes #1352

### CODE REVIEW

File: prov/gni/test/auth_key.c [hunk 1]
```diff
@@ -76,3 +76,42 @@ Test(auth_key, failed_insert)
...
+
+	auth_key = GNIX_GET_AUTH_KEY(NULL, 0);
+	cr_assert_neq(auth_key, NULL, "failed to get authorization key");
+
+	return NULL;
+}
+
+Test(auth_key, race_create)
+{
+#define __AUTH_KEY_THREAD_COUNT 8
```
Comment (sungeunchoi) [inline prov/gni/test/auth_key.c]: MOAR THREADS!!1!

Comment (sungeunchoi) [reply prov/gni/test/auth_key.c]: Seriously.  Make like 47 (a nice prime number).

File: prov/gni/test/auth_key.c [hunk 2]
```diff
@@ -76,3 +76,42 @@ Test(auth_key, failed_insert)
...
+	cr_assert_eq(ret, 0, "failed to initialize barrier");
+
+	for (i = 0; i < thread_count; i++) {
+		ret = pthread_create(&threads[i], NULL,
+			race_create_func, &barrier);
+		cr_assert_eq(ret, 0, "failed to create pthread");
+	}
+
+	for (i = 0; i < thread_count; i++) {
+		ret = pthread_join(threads[i], NULL);
```
Comment (sungeunchoi) [inline prov/gni/test/auth_key.c:120]: Check return code with cr_assert or something.

File: prov/gni/test/auth_key.c [hunk 3]
```diff
@@ -76,3 +76,42 @@ Test(auth_key, failed_insert)
...
+	}
+
+	ret = pthread_barrier_destroy(&barrier);
```
Comment (sungeunchoi) [inline prov/gni/test/auth_key.c]: Check return code.

File: prov/gni/test/auth_key.c [hunk 4]
```diff
@@ -76,3 +76,42 @@ Test(auth_key, failed_insert)
 	ret = _gnix_auth_key_create(NULL, 0);
 	cr_assert(ret == NULL, "unexpectedly created auth_key");
 }
+
+void *race_create_func(void *context)
+{
+	pthread_barrier_t *barrier = (pthread_barrier_t *) context;
+	struct gnix_auth_key *auth_key;
+
+	pthread_barrier_wait(barrier);
```
Comment (sungeunchoi) [inline prov/gni/test/auth_key.c]: Check return code.  I know we're not testing pthreads, but if something goes wrong, more info is better for diagnosis.

**Comment by jswaro** [comment]
bot:retest

**Comment by jswaro** [comment]
bot:retest

**Comment by jswaro** [comment]
@sungeunchoi please re-review.

File: prov/gni/test/auth_key.c (hunk 4 above)
Comment (sungeunchoi) [reply prov/gni/test/auth_key.c]: You didn't check return code here as I asked.  Is there a reason why?

Comment (jswaro) [reply prov/gni/test/auth_key.c]: I missed it.

**Comment by jswaro** [comment]
@sungeunchoi please re-review.

**Review by sungeunchoi — APPROVED** [review]

**Pull request merged**